### PR TITLE
IL-4295

### PR DESCRIPTION
**Pull request by mario-ipor**

### CODE REVIEW

File: contracts/base/amm/AmmTreasuryBaseV1.sol [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+
 import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
 import "@openzeppelin/contracts/utils/math/SafeCast.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {IporMath} from "../../libraries/math/IporMath.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
```
Comment (pete-ipor) [inline contracts/base/amm/AmmTreasuryBaseV1.sol:12]: move up

File: contracts/base/amm/AmmTreasuryBaseV1.sol [hunk 2]
```diff
@@ -64,18 +69,19 @@ contract AmmTreasuryBaseV1 is
...
+                    balance.iporPublicationFee.toInt256() -
+                    balance.treasury.toInt256() -
+                        balance.totalLiquidationDepositBalance.toInt256()).toUint256();
 
         return liquidityPool;
     }
 
     function getVersion() external pure returns (uint256) {
-        return 2_001;
+        return 2_002;
```
Comment (pete-ipor) [inline contracts/base/amm/AmmTreasuryBaseV1.sol:84]: maybe we can change this to contract addrss?

File: contracts/base/amm/services/AmmOpenSwapServiceBaseV1.sol [hunk 3]
```diff
@@ -24,7 +24,7 @@ abstract contract AmmOpenSwapServiceBaseV1 {
 
     address public constant ETH_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
 
-    uint256 public immutable version = 1;
+    uint256 public immutable version = 2;
```
Comment (pete-ipor) [inline contracts/base/amm/services/AmmOpenSwapServiceBaseV1.sol:27]: maybe we can change this to contract address?

File: contracts/chains/arbitrum/router/IporProtocolRouterArbitrum.sol [hunk 4]
```diff
@@ -81,20 +85,20 @@ contract IporProtocolRouterArbitrum is IporProtocolRouterAbstract {
...
-                liquidityMiningLens: liquidityMiningLens,
-                powerTokenLens: powerTokenLens,
-                flowService: flowService,
-                stakeService: stakeService,
-                ammPoolsServiceWstEth: ammPoolsServiceWstEth,
-                ammPoolsLensWstEth: ammPoolsLensWstEth,
-                ammPoolsServiceUsdm: ammPoolsServiceUsdm,
-                ammPoolsLensUsdm: ammPoolsLensUsdm
-            });
+            ammSwapsLens: ammSwapsLens,
```
Comment (pete-ipor) [inline contracts/chains/arbitrum/router/IporProtocolRouterArbitrum.sol:88]: ask frontend if they use this ?

File: contracts/base/amm/services/AmmOpenSwapServiceBaseV1.sol (hunk 3 above)
Comment (mario-ipor) [reply contracts/base/amm/services/AmmOpenSwapServiceBaseV1.sol:27]: In router we are not using getVersion

File: contracts/chains/arbitrum/router/IporProtocolRouterArbitrum.sol (hunk 4 above)
Comment (maciek-ipor) [reply contracts/chains/arbitrum/router/IporProtocolRouterArbitrum.sol:90]: we are using `ammOpenSwapServiceWstEth` adddress field to read `openingFeeRate` and bunch of other params, how can we read it now? Should we take the address that is stored in "wstEth: wstEth" param?

**Review by pete-ipor — APPROVED** [review]

**Pull request merged**